README.md

```diff
@@ -162,6 +168,139 @@ Click on the button below to deploy this demo to the IBM Cloud.
 
 1. View the application online at the app URL, for example: https://my-app-name.mybluemix.net
 
+## Deploying to Openshift
+
+This code pattern can be deployed to any openshift cluster or [minishift](https://www.okd.io/minishift) running 3.2 or above.
+
+### Creating a project
+
+After logging in with `oc login`, ensure that you have a project set up. If not, create one as follows:
+
+```
+oc new-project nlu-project --display-name="NLU Code Pattern"
+```
+
+Ensure that your current project is set.
+
+```bash
+oc project nlu-project
+```
+
+### Automatic
+
+Run the deploy.sh script:
+
+```
+sh openshift/deploy.sh
+```
+
+The script will run until the app is deployed and available.
+
+### Manually
+
+<details>
+<summary>Manually deploy to openshift</summary>
+
+### Prepare the required parameters
```

Comment:
I think this should be a `####` header to match the other subtitles in the manual deployment section.